OutOfSchool/OutOfSchool.DataAccess/Repository/CodeficatorRepository.cs

```diff
@@ -42,9 +42,11 @@ public async Task<List<CodeficatorAddressDto>> GetFullAddressesByPartOfName(stri
                     from pp in db.CATOTTGs.Where(x2 => p.ParentId == x2.Id).DefaultIfEmpty()
                     from ppp in db.CATOTTGs.Where(x3 => pp.ParentId == x3.Id).DefaultIfEmpty()
                     from pppp in db.CATOTTGs.Where(x4 => ppp.ParentId == x4.Id).DefaultIfEmpty()
-                    where string.IsNullOrEmpty(namePart)
+                    where string.IsNullOrEmpty(namePart) && string.IsNullOrEmpty(e.Category)
                        ? EF.Property<bool>(e, "IsTop")
-                       : ((CodeficatorCategory.Level4.Name.Contains(e.Category) && e.Name.StartsWith(namePart)) || (e.Category == CodeficatorCategory.CityDistrict.Name && p.Name.StartsWith(namePart))) && categories.Contains(e.Category)
+                       : ((e.Name.StartsWith(namePart) &&
+                          (CodeficatorCategory.Level1.Name.Contains(e.Category) || CodeficatorCategory.Level4.Name.Contains(e.Category))) ||
+                          (e.Category == CodeficatorCategory.CityDistrict.Name && p.Name.StartsWith(namePart))) && categories.Contains(e.Category)
                     select new CodeficatorAddressDto
```

Comment:
Nope :)

Reply:
Smthn looks like this:
bool returnIsTop = string.IsNullOrEmpty(namePart) &&
(categories.Contains(CodeficatorCategory.SpecialStatusCity.Name) || categories.Contains(CodeficatorCategory.Region.Name));

......

where returnIsTop
? EF.Property(e, "IsTop")
.......

Reply:
Fixed